### ISSUE TITLE
PowerShellTools fails to initialize

### ISSUE THREAD

**Issue by mirola** [adamdriscoll/poshtools#160]
Hi,

PowerShellTools fails to initialize any time i start VS.

The only error message i get is a MessageBox:

![powershelltoolserror](https://cloud.githubusercontent.com/assets/10462370/5679083/0a45ec0a-97fa-11e4-86a1-863fffbe5933.PNG)

There is no error in VS ActivityLog.xml.

Changing the Powesrhell ExecutionPolicy from AllSigned to Bypass or Unrestricted in any scope has no effect.

Windows 8.1 Pro, VS 2013 Update 4, PowerShell Tools for Visual Studio 2013 1.0.5

Any y help would be appreciated

Mirola

**Comment by adamdriscoll** [comment]
Very weird stack trace. I'm asking around to see if anyone else has seen this.

**Comment by MaxTrinidad** [comment]
Is this issue possibly on non-English version.  I've been using it for awhile with the VS 2013 Community version and SQL Data Tools. 

Strange! 

One thing to point out!!  Just to make sure to open VS "as Administrator".

:)
Max

**Comment by adamdriscoll** [comment]
It was also mentioned that it may have to do with one of the types.ps1xml files that you have loaded in your session. Could you provide a list of modules you have installed?

**Comment by mirola** [comment]
Yes win 8.1 and vs are german verisons.

It doesn't matter if i open vs as administrator or if i open vs as a normal user. In both cases Powershell Tools raises that exception.

The only powershell modules loaded in a standard PS session as user are:
32 Bit PS: *Microsoft.PowerShell.Management*
64 Bit PS: *Microsoft.PowerShell.Management* and *Microsoft.PowerShell.Utility*

Here is the list of modules installed:
*AppBackgroundTask
AppLocker
Appx
AssignedAccess
BitLocker
BitsTransfer
BranchCache
CimCmdlets
Defender
DirectAccessClientComponents
Dism
DnsClient
Hyper-V
International
iSCSI
ISE
Kds
Microsoft.PowerShell.Diagnostics
Microsoft.PowerShell.Host
Microsoft.PowerShell.Management
Microsoft.PowerShell.Security
Microsoft.PowerShell.Utility
Microsoft.WSMan.Management
MMAgent
MsDtc
NetAdapter
NetConnection
NetEventPacketCapture
NetLbfo
NetNat
NetQos
NetSecurity
NetSwitchTeam
NetTCPIP
NetWNV
NetworkConnectivityStatus
NetworkTransition
PcsvDevice
PKI
PrintManagement
PSDesiredStateConfiguration
PSDiagnostics
PSScheduledJob
PSWorkflow
PSWorkflowUtility
ScheduledTasks
SecureBoot
SmbShare
SmbWitness
StartScreen
Storage
TLS
TroubleshootingPack
TrustedPlatformModule
VpnClient
Wdac
WebAdministration
WindowsDeveloperLicense
WindowsErrorReporting
WindowsSearch
SQLASCMDLETS
SQLPS*

I did some more investigations in that PS Tools error. I find Nuget fails while opening its console (a PS-Console). Error from VS ActivityLog.xml (translated from german):
*<entry>
    <record>575</record>
    <time>2015/01/05 22:19:07.397</time>
    <type>Error</type>
    <source>NuGet Package Manager</source>
    <description>Object reference not set to an instance of an object.   at Microsoft.PowerShell.Commands.Internal.Format.XmlLoaderBase.MatchNodeNameHelper(XmlNode n, String s, Boolean allowAttributes)
   at Microsoft.PowerShell.Commands.Internal.Format.TypeInfoDataBaseLoader.LoadData(XmlDocument doc, TypeInfoDataBase db)
   at Microsoft.PowerShell.Commands.Internal.Format.TypeInfoDataBaseLoader.LoadXmlFile(XmlFileLoadInfo info, TypeInfoDataBase db, MshExpressionFactory expressionFactory, AuthorizationManager authorizationManager, PSHost host, Boolean preValidated)
   at Microsoft.PowerShell.Commands.Internal.Format.TypeInfoDataBaseManager.LoadFromFileHelper(Collection`1 files, MshExpressionFactory expressionFactory, AuthorizationManager authorizationManager, PSHost host, Boolean preValidated, List`1& logEntries, Boolean& success)
   at Microsoft.PowerShell.Commands.Internal.Format.TypeInfoDataBaseManager.LoadFromFile(Collection`1 files, MshExpressionFactory expressionFactory, Boolean acceptLoadingErrors, AuthorizationManager authorizationManager, PSHost host, Boolean preValidated, List`1& logEntries)
   at Microsoft.PowerShell.Commands.Internal.Format.TypeInfoDataBaseManager.UpdateDataBase(Collection`1 mshsnapins, AuthorizationManager authorizationManager, PSHost host, Boolean preValidated)
   at System.Management.Automation.Runspaces.InitialSessionState.UpdateFormats(ExecutionContext context, Boolean update)
   at System.Management.Automation.Runspaces.InitialSessionState.Bind(ExecutionContext context, Boolean updateOnly, PSModuleInfo module, Boolean noClobber, Boolean local)
   at System.Management.Automation.AutomationEngine..ctor(PSHost hostInterface, RunspaceConfiguration runspaceConfiguration, InitialSessionState iss)
   at System.Management.Automation.Runspaces.LocalRunspace.DoOpenHelper()
   at System.Management.Automation.Runspaces.LocalRunspace.OpenHelper(Boolean syncCall)
   at System.Management.Automation.Runspaces.RunspaceBase.CoreOpen(Boolean syncCall)
   at System.Management.Automation.Runspaces.RunspaceBase.Open()
   at NuGetConsole.Host.PowerShell.Implementation.RunspaceManager.CreateRunspace(IConsole console, String hostName)
   at NuGetConsole.Host.PowerShell.Implementation.RunspaceManager.CreateAndSetupRunspace(IConsole console, String hostName)
   at NuGetConsole.Host.PowerShell.Implementation.RunspaceManager.<>c__DisplayClass0.<GetRunspace>b__2(String name)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at NuGetConsole.Host.PowerShell.Implementation.RunspaceManager.GetRunspace(IConsole console, String hostName)
   at NuGetConsole.Host.PowerShell.Implementation.PowerShellHost.Initialize(IConsole console)</description>
  </entry>*

Doing some deeper investigations on that nuget error leads me to the follwing error message of Sharpdevelop nuget console (translated from german):

*Object reference not set to an instance of an object.
Cannot invoke pipeline because runspace is not in the Opened state. Current state of runspace is ‘Broken’.
Cannot invoke pipeline because runspace is not in the Opened state. Current state of runspace is ‘Broken’.
Cannot invoke pipeline because runspace is not in the Opened state. Current state of runspace is ‘Broken’.
Cannot invoke pipeline because runspace is not in the Opened state. Current state of runspace is ‘Broken’.
Cannot invoke pipeline because runspace is not in the Opened state. Current state of runspace is ‘Broken’.*

So it seems to me that the call of *RunspaceFactory.CreateRunspace()* somewhere in Powershell Tools **and** Nuget code returns a runspace being in a broken state. So the *Open()* call fails. I don't have any idea what causes that.

**Comment by kemmis** [comment]
1.0.5 here - this is the error I get w/ VS 2013 Premium Update 4

![image](https://cloud.githubusercontent.com/assets/647545/5723158/f25d8382-9b0f-11e4-8247-8556b6872b2f.png)

**Comment by adamdriscoll** [comment]
@kemmis - You need to make sure you have Windows Management Framework v3 installed. 
@mirola - That's really interesting. I have no idea what would be causing that. I might need to do some testing on a German OS and take a closer look at those modules you listed.

**Comment by AndreSayreMSFT** [comment]
@mirola If I'm reading this correctly, it crashes when launching the NuGet package manager, and without PoshTools?  That's what I'm getting from the Stack Trace at least!  Does the PowerShell ISE error out for you as well?

**Comment by mirola** [comment]
@Andre Sayre: It does not matter if Nuget is installed or not - PoshTools fails always. Same with NuGet, NuGet console crashes - with or without PoshTools installed. PS-ISE wors like a charm.

**Comment by adamdriscoll** [comment]
That's really weird. Have you tried the x86 ISE?

**Comment by AndreSayreMSFT** [comment]
@mirola Would you be willing to share your .ps1xml files?  Our friends within the PowerShell team suspect there's something there that would bear investigation for this issue.

**Comment by gerane** [comment]
I am getting this issue in the following setup.

Windows 8.1 Pro, 
VS 2012 Update 4, 
PowerShell Tools for Visual Studio 2012 1.0.5

**Comment by AndreSayreMSFT** [comment]
@gerane Does this show up with latest [dev builds](https://ci.appveyor.com/project/PoshTools-MSFT/poshtools/branch/dev/artifacts) for you still? You may need to uninstall the current version of the tools and reinstall.

We're unable to reproduce this still on our side, and hope to get to the bottom of this!

**Comment by tpoche** [comment]
I was getting a similar error everytime I ran Visual Studio 2013 after installing the extension.  Turns out this was being caused by the fact that I had installed the extension without a recent version of PowerShell installed.  This error would persist even after I installed Windows Management Framework 4.0.  I only got it working after uninstalling and reinstalling the extension with the Windows Management Framework 4.0 <i>already</i> installed.  Hope this helps some of you seeing the same issue.
